readthedocs/proxito/views/hosting.py

```diff
@@ -711,6 +722,31 @@ def _get_filetreediff_response(self, *, request, project, version, resolver):
             },
         }
 
+    def _get_filesections_response(self, *, project, version):
+        """
+        Get the file sections response for the given version.
+        """
+        manifest = get_section_manifest(version)
+        if not manifest:
+            return None
+
+        return {
```

Comment:
Example in dev from Addons. 

```
{
    "enabled": true,
    "sections": [
        {
            "path": "genindex.html",
            "sections": [
                {
                    "id": "index",
                    "title": "Index"
                }
            ]
        },
        {
            "path": "index.html",
            "sections": [
                {
                    "id": "read-the-docs-addons",
                    "title": "🦸 Read the Docs Addons"
                },
                {
                    "id": "examples",
                    "title": "Examples"
                },
                {
                    "id": "contents",
                    "title": "Contents"
                }
            ]
        },
        {
            "path": "releasing.html",
            "sections": [
                {
                    "id": "releasing",
                    "title": "Releasing"
                }
            ]
        },
        {
            "path": "search.html",
            "sections": [
                {
                    "id": "",
                    "title": "Search - Read the Docs Addons documentation"
                }
            ]
        },
        {
            "path": "testing.html",
            "sections": [
                {
                    "id": "testing",
                    "title": "Testing"
                },
                {
                    "id": "localhost",
                    "title": "localhost"
                },
                {
                    "id": "read-the-docs",
                    "title": "Read the Docs"
                }
            ]
        }
    ]
}
```

Reply:
We shouldn't return the content of manifest here, but instead, we should return the URL where the frontend would download it and parse it on the client side. Otherwise, this response will be huge and most of the times it's not going to be used.

Reply:
By the way, I think we will probably want to do the same for filetreediff as well; since there will be PRs will a ton of files changed.

Reply:
Yea I just copied the implantation from filetreediff. I figured in production we would only pull the data when somebody pulls up the command pallet, but this made it easy to test.

In theory, the data is also already in our search index, I just wasn't 100% sure how to get it out of elastic.